test/components/connect.spec.js

```diff
@@ -480,6 +480,37 @@ describe('React', () => {
       expect(stub.props.stateThing).toBe('HELLO azbzcZ')
     })
 
+    it('should allow providing a factory function to mergeProps', () => {
+      const store = createStore(() => ({
+        a: 123
+      }))
+
+      function sum(a, b) {
+        return a + b
+      }
+
+      @connect(
+        state => state,
+        () => ({ sum }),
+        () => (stateProps, dispatchProps, parentProps) => ({
```

Comment:
The test should probably cover that the factory part of the mergeProps method is only called once, related to my previous comment.